File: frontend/SUT/src/interface/Concert.ts
// import { UsersInterface } from './IUser';
// import { SeatChart } from './seatChart';
// import { PerformanceSchedule } from './performanceSchedule';

export interface ConcertInterface {
  ID?: number; 
  Name?: string;
  ShowDate?: Date; 
  Artist?: string;
  Description?: string;
  BenefitFile?: File; 
  InfoFile?: File;
  PosterFile?: File;
  TicketSalesStartDate?: Date;
  Venue?: string;
  Status?: string;

  // organizerID?: number; 
  // organizer?: UsersInterface; 

  // seatCharts?: SeatChart[]; // การใช้ `?` แสดงว่าเป็น optional
  // performanceSchedules?: PerformanceSchedule[]; // การใช้ `?` แสดงว่าเป็น optional
}

File: backend/entity/concert.go
package entity

import (
    "time"
    "gorm.io/gorm"
)

type Concert struct {
    gorm.Model
    Name                    string    `json:"name"`                     
    ShowDate                time.Time `json:"show_date"`                
    Artist                  string    `json:"artist"`                   
    Description             string    `json:"description"`             
	PosterFile              string    `json:"poster_file"`   
	BenefitFile             string    `json:"benefit_file"` 
	InfoFile                string    `json:"info_file"`             
    TicketSalesStartDate    time.Time `json:"ticket_sales_start_date"`  
    Venue                   string    `json:"venue"`                    
    Status                  string    `json:"status"`        

    OrganizerID             *uint     
    Organizer               Organizer   `gorm:"foreignKey:OrganizerID"`
    SeatCharts              []SeatChart `gorm:"foreignKey:ConcertID"`
    PerformanceSchedules     []PerformanceSchedule `gorm:"foreignKey:ConcertID"`
}
